### PR TITLE
Update cosign to > v1.12.1 and sigstore to v1.4.2

### PR DESCRIPTION
**Pull request by RTann**
## Checklist
- [x] Investigated and inspected CI test results
- ~[ ] Unit test and regression tests added~
- ~[ ] Evaluated and added CHANGELOG entry if required~
- ~[ ] Determined and documented upgrade steps~
- ~[ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))~

## Testing Performed

CI

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -444,17 +443,6 @@ replace (
 	// github.com/stackrox/helm-operator is a modified fork of github.com/operator-framework/helm-operator-plugins that
 	// we currently depend on.
 	github.com/operator-framework/helm-operator-plugins => github.com/stackrox/helm-operator v0.0.10-0.20220919093109-89f9785764c6
-	// github.com/sigstore/rekor is a transitive dep pulled in by cosign. The version pulled in by cosign is using
```
Comment (RTann) [inline go.mod:447]: Note: the newer versions of cosign and sigstore use go-tuf >= v0.5.0

**Review by dhaus67 — APPROVED** [review]

**Comment by RTann** [comment]
Update: https://github.com/sigstore/cosign/issues/2277 has been resolved and will be released in v1.13.0, which I'd expect to be released sometime soon. I will wait until that is released before updating this PR and merging

File: go.mod [hunk 2]
```diff
@@ -308,25 +311,25 @@ require (
 	github.com/russross/blackfriday v1.6.0 // indirect
 	github.com/russross/blackfriday/v2 v2.1.0 // indirect
 	github.com/sassoftware/relic v0.0.0-20210427151427-dfb082b79b74 // indirect
-	github.com/secure-systems-lab/go-securesystemslib v0.3.1 // indirect
+	github.com/secure-systems-lab/go-securesystemslib v0.4.0 // indirect
 	github.com/shibumi/go-pathspec v1.3.0 // indirect
 	github.com/shopspring/decimal v1.2.0 // indirect
-	github.com/sigstore/rekor v0.4.1-0.20220114213500-23f583409af3 // indirect
+	github.com/sigstore/rekor v0.12.1-0.20220915152154-4bb6f441c1b2 // indirect
```
Comment (RTann) [inline go.mod:317]: This version of rekor uses go-tuf v0.5.0, which has the fix for https://github.com/theupdateframework/go-tuf/security/advisories/GHSA-66x3-6cw3-v5gj

https://github.com/sigstore/rekor/blob/4bb6f441c1b27ccc7e625c721c7d3203acc7b313/go.mod

File: go.mod [hunk 3]
```diff
@@ -308,25 +311,25 @@ require (
...
 	github.com/steveyen/gtreap v0.1.0 // indirect
 	github.com/stretchr/objx v0.4.0 // indirect
-	github.com/subosito/gotenv v1.3.0 // indirect
+	github.com/subosito/gotenv v1.4.1 // indirect
 	github.com/syndtr/gocapability v0.0.0-20200815063812-42c35b437635 // indirect
-	github.com/syndtr/goleveldb v1.0.1-0.20210819022825-2ae1ddf74ef7 // indirect
+	github.com/syndtr/goleveldb v1.0.1-0.20220721030215-126854af5e6d // indirect
 	github.com/tent/canonical-json-go v0.0.0-20130607151641-96e4ba3a7613 // indirect
-	github.com/theupdateframework/go-tuf v0.3.0 // indirect
+	github.com/theupdateframework/go-tuf v0.5.1-0.20220920170306-f237d7ca5b42 // indirect
```
Comment (RTann) [inline go.mod:332]: we even use a version > v0.2.0 explicitly here

**Comment by RTann** [comment]
I wanted to merge this asap so we can have the vuln fix asap. I imagine cosign v1.13.0 should come out before we cut a new release, so we will update to that before we cut the new release

**Pull request merged**